js/textInputModel.js

```diff
@@ -185,6 +184,20 @@ class TextInputModel extends QuestionModel {
     return userAnswer.trim();
   }
 
+  getAnswerIndex(answers, userAnswer) {
+    const cleanUserAnswer = this.cleanupUserAnswer(userAnswer);
+
+    if (!this.get('_allowsAnyCase')) {
+      return answers.indexOf(cleanUserAnswer);
+    }
+
+    const matches = answers.find(answer => {
+      return this.cleanupUserAnswer(answer) === cleanUserAnswer;
+    });
+
+    return answers.indexOf(matches);
+  }
```

Comment:
```suggestion
  getAnswerIndex(answers, userAnswer) {    
    const cleanAnswers = answers.map(this.cleanupAnswer);
    const cleanUserAnswer = this.cleanupAnswer(userAnswer);
    return cleanAnswers.findIndex(cleanAnswer => cleanAnswer === cleanUserAnswer);
  }
```

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex


Reasons for suggestion:

```js
  getAnswerIndex(answers, userAnswer) {
    const cleanUserAnswer = this.cleanupUserAnswer(userAnswer);

    if (!this.get('_allowsAnyCase')) { // is this needed? cleanupUserAnswer does this already
      return answers.indexOf(cleanUserAnswer);
    }

    // array.prototype has findIndex, we shouldn't need to do find then indexOf
    // matches should be match as it's not an array, find always returns a single item
    const matches = answers.find(answer => { 
      return this.cleanupUserAnswer(answer) === cleanUserAnswer;
    });

    return answers.indexOf(matches); 
  }
```

Reply:
Removed this lookup in favour of a simplified map approach